src/components/smart_objects/include/smart_objects/number_schema_item.h

```diff
@@ -148,10 +159,26 @@ Errors::eType TNumberSchemaItem<NumberType>::validate(
 
   NumberType rangeLimit;
   if (mMinValue.getValue(rangeLimit) && (value < rangeLimit)) {
+    if (!Object.getKey().empty()) {
+      errorMessage.assign("Validation failed for \"" + Object.getKey() +
+                          "\". ");
+    }
+    std::stringstream stream;
+    stream << "Value too small, got: " << value
+           << ", minimum allowed: " << rangeLimit;
+    errorMessage += stream.str();
     return Errors::OUT_OF_RANGE;
   }
 
   if (mMaxValue.getValue(rangeLimit) && (value > rangeLimit)) {
+    if (!Object.getKey().empty()) {
+      errorMessage.assign("Validation failed for \"" + Object.getKey() +
+                          "\". ");
+    }
+    std::stringstream stream;
+    stream << "Value too large, got: " << value
+           << ", maximum allowed: " << rangeLimit;
+    errorMessage += stream.str();
```

Comment:
@jacobkeeler looks like errorMessage used something like a recursive validation report? std::string is not best container for such purpose. We have such validation type in policy table, you can do something the same.